### ISSUE TITLE
False positive and bad autocorrect in Style/TrailingUnderscoreVariable

### ISSUE THREAD

**Issue by mvz** [bbatsov/rubocop#4757]
For complex destructuring assignments, Style/TrailingUnderscoreVariable gives false positives and performs the wrong autocorrect.

## Expected behavior

Given the following code:
```ruby
a = [1, 2, [3, 4, 5]]
_, b, (_, c,) = a
_, d, (_, e, _) = a

p a, b, c, d, e
```

I expect RuboCop to:
* report no offense on the second line
* report an offense on the third line, suggesting a change to `_, d, (_, e,) = a`
* autocorrect the third line to  `_, d, (_, e,) = a`

## Actual behavior

RuboCop reports:
```
foo.rb:2:7: C: Do not use trailing _s in parallel assignment. Prefer _, b, = a.
_, b, (_, c,) = a
      ^^^^^^^^
foo.rb:3:7: C: Do not use trailing _s in parallel assignment. Prefer _, d, = a.
_, d, (_, e, _) = a
      ^^^^^^^^^^
```

RuboCop autocorrects the code to:
```ruby
a = [1, 2, [3, 4, 5]]
_, b, = a
_, d, = a

p a, b, c, d, e
```

## Steps to reproduce the problem

Given a file `foo.rb` with:

```ruby
a = [1, 2, [3, 4, 5]]
_, b, (_, c,) = a
_, d, (_, e, _) = a

p a, b, c, d, e
```

* run `rubocop foo.rb` to show the false positive
* run `rubocop -a foo.rb` to show the broken autocorrect

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```
